sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBoundAttribute.scala

```diff
@@ -106,4 +108,12 @@ case class GpuBoundReference(ordinal: Int, dataType: DataType, nullable: Boolean
       case cv: GpuColumnVector => cv.incRefCount()
     }
   }
+
+  override def convertToAst(numFirstTableColumns: Int): ast.AstNode = {
+    if (ordinal >= numFirstTableColumns) {
```

Comment:
We're currently passing `Int.MaxValue` as the number of columns when there's only one table, so we can trigger off of that if we ever find this is useful in practice.

Reply:
`Int.MaxValue` does what I would want, assuming cudf throws an exception if not enough parameters are passed in.

Reply:
> assuming cudf throws an exception if not enough parameters are passed in.

cudf does not throw an exception.  It silently clamps column indices to the input table and maps all other table references to the left input table.

Reply:
It is still fine for now. Just would be nice to get some kind of an actual error if we did something wrong. That way in some cases we can debug an exception says your expression does not look correct instead of an exception that says the data didn't match, or he data types for a column didn't match.